src/meta-srv/src/selector/load_based.rs

```diff
@@ -143,10 +143,14 @@ async fn get_leader_peer_ids(
         .context(error::TableMetadataManagerSnafu)
         .map(|route| {
             route.map_or_else(Vec::new, |route| {
-                find_leaders(route.region_routes())
-                    .into_iter()
-                    .map(|peer| peer.id)
-                    .collect()
+                find_leaders(
+                    route
+                        .region_routes()
+                        .expect("expected physical table route"),
```

Comment:
The "expect" is not safe here. You can `ensure!` the table route is of type physical above first.